### PR TITLE
Update `ed25519-dalek` & `crypto_box`

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

**Comment by mmmarcos** [comment]
Are all changes necessary to update `ed25519-dalek`? Can you summarize or briefly explain the changes included in the PR?

**Comment by FirelightFlagboy** [comment]
@mmmarcos I've added a description to each commit (minus the last one)

**Comment by mmmarcos** [comment]
> @mmmarcos I've added a description to each commit (minus the last one)

@FirelightFlagboy thanks, much clearer now! The only thing missing seems to be the "fast" feature [here](https://github.com/Scille/parsec-cloud/pull/5621/commits/5b4a600eb48efdbdbd4379db2f2583ead78e3ff9) which does not seem to be documented in https://github.com/dalek-cryptography/curve25519-dalek/tree/main/ed25519-dalek

**Review by mmmarcos — APPROVED** [review]
LGTM

File: libparsec/crates/crypto/src/rustcrypto/private.rs [hunk 1]
```diff
@@ -134,13 +134,11 @@ impl PrivateKey {
     }
 
     pub fn generate_shared_secret_key(&self, peer_public_key: &PublicKey) -> SecretKey {
-        SecretKey::from(x25519(*self.0.as_bytes(), *peer_public_key.0.as_bytes()))
+        SecretKey::from(x25519(self.0.to_bytes(), peer_public_key.0.to_bytes()))
```
Comment (TimeEngineer) [inline libparsec/crates/crypto/src/rustcrypto/private.rs:137]: why this change ?
`to` is supposed to be slow instead of `as`

Comment (TimeEngineer) [reply libparsec/crates/crypto/src/rustcrypto/private.rs:137]: And you added the method for `self`, so you don't need to call `.0`

Comment (FirelightFlagboy) [reply libparsec/crates/crypto/src/rustcrypto/private.rs:137]: Read the commit description

**Review by TimeEngineer — APPROVED** [review]
